qiskit/synthesis/discrete_basis/commutator_decompose.py

```diff
@@ -92,8 +92,8 @@ def _solve_decomposition_angle(matrix: np.ndarray) -> float:
     lhs = math.sin(angle / 2)
 
     def objective(phi):
-        sin_sq = np.sin(phi / 2) ** 2
-        return 2 * sin_sq * np.sqrt(1 - sin_sq**2) - lhs
+        sin_sq = math.sin(phi[0] / 2) ** 2
+        return 2 * sin_sq * math.sqrt(1 - sin_sq**2) - lhs
```

Comment:
This looks like a behavioural change - how sure are we that `phi` is a single item?  Perhaps `phi.item()` is better if we're sure, because it'll fail noisily if it's not.

Reply:
we are sure its a single item as its only called from `fsolve(objective, angle)` where `angle` is scalar, but I changed to `phi.item()` for ease of future understanding in af57ee4cb4c421d922c16b22cf69a1bc85efdf98

Reply:
I'm kind of surprised that `fsolve` allows the objective function to return a differently-shaped array to what it passes in, but as long as it works, I'm good with it.